### PR TITLE
add code to skip adding debian-security on raspberrypi boards

### PR DESCRIPTION
**Pull request by ryecoaaron**
Skip adding debian security on Raspbian

Raspbian and RaspberryPi OS (not 64bit) rebuild packages from debian-security.  If they do not change the version and the debian-security repo is enabled, apt will have a package checksum mismatch.

Fixes: package checksum mismatches

Signed-off-by: Aaron Murray <plugins@omv-extras.org>
"""

The Signed-off-by line is important, and it is your certification that your contributions satisfy the developers certificate or origin.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview. More information for contributors is available here:
https://docs.openmediavault.org/en/latest/development/contribute.html
-->

- [ ] References issue
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: deb/openmediavault/srv/salt/omv/deploy/apt/default.sls [hunk 1]
```diff
@@ -20,6 +20,7 @@
 {% set config = salt['omv_conf.get']('conf.system.apt.distribution') %}
 {% set use_kernel_backports = salt['pillar.get']('default:OMV_APT_USE_KERNEL_BACKPORTS', True) -%}
 {% set proxy_config = salt['omv_conf.get']('conf.system.network.proxy') %}
+{% set rpi = salt['pillar.get']('default:OMV_IS_RPI', False) %}
```
Comment (votdev) [inline deb/openmediavault/srv/salt/omv/deploy/apt/default.sls:23]: Could we name this variable like this? Then it is equal to `OMV_APT_USE_KERNEL_BACKPORTS`.
```suggestion
{% set use_os_security = salt['pillar.get']('default:OMV_APT_USE_OS_SECURITY', False) %}
```

**Review by votdev — CHANGES_REQUESTED** [review]

**Comment by votdev** [comment]
Uahhh, i really hate such workarounds. To me it looks like RaspberryPi upstream is doing something terrible wrong.

**Comment by ryecoaaron** [comment]
I understand but I don’t have a better solution.

File: deb/openmediavault/srv/salt/omv/deploy/apt/default.sls [hunk 2]
```diff
@@ -20,6 +20,7 @@
 {% set config = salt['omv_conf.get']('conf.system.apt.distribution') %}
 {% set use_kernel_backports = salt['pillar.get']('default:OMV_APT_USE_KERNEL_BACKPORTS', True) -%}
 {% set proxy_config = salt['omv_conf.get']('conf.system.network.proxy') %}
+{% set use_os_security = salt['pillar.get']('default:OMV_APT_USE_OS_SECURITY', False) %}
```
Comment (votdev) [inline deb/openmediavault/srv/salt/omv/deploy/apt/default.sls:23]: Could you please set this to true? On a Raspberry installation this env variable must be set explicitly.

**Review by votdev — CHANGES_REQUESTED** [review]

File: deb/openmediavault/srv/salt/omv/deploy/apt/default.sls [hunk 3]
```diff
@@ -110,6 +111,8 @@ remove_apt_sources_list_security_{{ loop.index0 }}:
       - repo: "{{ security_pkg_repo.line }}"
 {% endfor %}
 
+{% if not use_os_security | to_bool %}
```
Comment (votdev) [inline deb/openmediavault/srv/salt/omv/deploy/apt/default.sls:114]:
```suggestion
{% if use_os_security | to_bool %}
```

File: deb/openmediavault/srv/salt/omv/deploy/apt/default.sls (hunk 2 above)
Comment (votdev) [inline deb/openmediavault/srv/salt/omv/deploy/apt/default.sls:23]:
```suggestion
{% set use_os_security = salt['pillar.get']('default:OMV_APT_USE_OS_SECURITY', True) %}
```

**Review by votdev — CHANGES_REQUESTED** [review]

**Review by votdev — APPROVED** [review]

**Comment by votdev** [comment]
@ryecoaaron I stumbled over this code [here](https://github.com/openmediavault/openmediavault/blob/5.x/deb/openmediavault/srv/salt/omv/deploy/apt/default.sls#L105) and wonder why your PR is necessary. Salt should remove the `debian-security` repositories spread across other APT config files, so only the `/etc/apt/sources.list.d/openmediavault-debian-security.list` should exist afterwards.

**Comment by rabelux** [comment]
This feature has been released, right?
Is there already an implementation of setting the env-var?

I ask because on my raspberry pi (with OMV 5.6.7) the security-repo got readded yesterday. I had removed it about a month ago.

`omv-env list` shows that the variable `OMV_APT_USE_OS_SECURITY` is known, but has no value as
`omv-env get OMV_APT_USE_OS_SECURITY` returns nothing.

I guess l can set this variable to False with `omv-env set OMV_APT_USE_OS_SECURITY False` ?

**Comment by ryecoaaron** [comment]
Yes, it has been released.  The only thing that sets it is the install script.  If you have an existing install, you will need to run the omv-env command to set it.  omv-env set OMV_APT_USE_OS_SECURITY False

**Comment by rabelux** [comment]
Did that, thanks for your response 👍

**Pull request merged**